fe/fe-core/src/main/java/com/starrocks/transaction/TransactionStmtExecutor.java

```diff
@@ -0,0 +1,499 @@
+import com.starrocks.common.TimeoutException;
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.common.util.concurrent.lock.LockTimeoutException;
+import com.starrocks.lake.LakeTableHelper;
+import com.starrocks.load.EtlJobType;
+import com.starrocks.load.loadv2.InsertLoadJob;
+import com.starrocks.load.loadv2.LoadJob;
+import com.starrocks.load.loadv2.LoadMgr;
+import com.starrocks.metric.MetricRepo;
+import com.starrocks.metric.TableMetricsEntity;
+import com.starrocks.metric.TableMetricsRegistry;
+import com.starrocks.planner.FileScanNode;
+import com.starrocks.planner.OlapScanNode;
+import com.starrocks.planner.ScanNode;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.DefaultCoordinator;
+import com.starrocks.qe.DmlType;
+import com.starrocks.qe.OriginStatement;
+import com.starrocks.qe.QeProcessorImpl;
+import com.starrocks.qe.SessionVariable;
+import com.starrocks.qe.scheduler.Coordinator;
+import com.starrocks.rpc.RpcException;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.service.FrontendOptions;
+import com.starrocks.sql.ast.CreateTableAsSelectStmt;
+import com.starrocks.sql.ast.DeleteStmt;
+import com.starrocks.sql.ast.DmlStmt;
+import com.starrocks.sql.ast.InsertStmt;
+import com.starrocks.sql.ast.StatementBase;
+import com.starrocks.sql.ast.UpdateStmt;
+import com.starrocks.sql.ast.txn.BeginStmt;
+import com.starrocks.sql.ast.txn.CommitStmt;
+import com.starrocks.sql.ast.txn.RollbackStmt;
+import com.starrocks.sql.plan.ExecPlan;
+import com.starrocks.task.LoadEtlTask;
+import com.starrocks.thrift.TExplainLevel;
+import com.starrocks.thrift.TLoadJobType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.starrocks.common.ErrorCode.ERR_TXN_NOT_EXIST;
+
+public class TransactionStmtExecutor {
+    private static final Logger LOG = LogManager.getLogger(TransactionStmtExecutor.class);
+
+    public static void beginStmt(ConnectContext context, BeginStmt stmt) {
+        if (context.getExplicitTxnState() != null) {
+            //Repeated begin does not create a new transaction
+            ExplicitTxnState explicitTxnState = context.getExplicitTxnState();
+            String label = explicitTxnState.getTransactionState().getLabel();
+            long transactionId = explicitTxnState.getTransactionState().getTransactionId();
+            context.getState().setOk(0, 0, buildMessage(label, TransactionStatus.PREPARE, transactionId, -1));
+            return;
+        }
+
+        long transactionId = GlobalStateMgr.getCurrentState().getGlobalTransactionMgr()
+                .getTransactionIDGenerator().getNextTransactionId();
+        String label = DebugUtil.printId(context.getExecutionId());
+        TransactionState transactionState = new TransactionState(transactionId, label, null,
+                TransactionState.LoadJobSourceType.INSERT_STREAMING,
```

Comment:
what does "STREAMING" mean?

Reply:
For the purposes of this PR, it does not make sense. This is for historical reasons, insert was called "INSERT_STREAMING"， Because there was a not streamming insert...